commit f56c51593d7b163126adc154d739ec813fc296aa
Author: Umaro <umaro@narshe.net>
Date:   Tue Jun 21 19:43:00 2011 +0200

    mmc: Attribute the IO wait time properly in mmc_wait_for_req().
    
        In mmc_wait_for_req() function, change the call wait_for_completion()
        to wait_for_completion_io(). This change makes the kernel account for
        wait time as I/O wait and through another configuration this I/O wait
        is treated as busy which makes the acpu clock to scale up.
    
        CRs-Fixed: 283420
        Change-Id: If3de6325b7e4cf8141c808e5853f4fc2dbb4e3ec
        Signed-off-by: Murali Palnati <palnatim@codeaurora.org>

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index cb6fd64..17bf31c 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -254,7 +254,7 @@ void mmc_wait_for_req(struct mmc_host *host, struct mmc_request *mrq)
 	} else
 #endif
 #endif
-		wait_for_completion(&complete);
+		wait_for_completion_io(&complete);
 
 }
 
